cg/cli/clean.py

```diff
@@ -106,46 +108,68 @@ def scout_finished_cases(
         context.invoke(hk_alignment_files, bundle=bundle, yes=yes, dry_run=dry_run)
 
 
-@clean.command("hk-past-files")
-@click.option("-c", "--case-id", type=str)
-@click.option("-t", "--tags", multiple=True)
-@click.option("-y", "--yes", is_flag=True, help="skip checks")
+@clean.command("hk-bundle-files")
+@click.option("-c", "--case_id", type=str, required=False)
+@click.option("-p", "--pipeline", type=Pipeline, required=False)
+@click.option("-t", "--tags", multiple=True, required=True)
+@click.option("-o", "--days-old", type=int, default=30)
 @click.option("-d", "--dry-run", is_flag=True, help="Shows cases and files that would be cleaned")
 @click.pass_obj
-def hk_past_files(context: CGConfig, case_id: str, tags: list, yes: bool, dry_run: bool):
-    """Remove files found in older housekeeper bundles"""
+def hk_bundle_files(
+    context: CGConfig,
+    case_id: Optional[str],
+    tags: list,
+    days_old: Optional[int],
+    pipeline: Optional[Pipeline],
+    dry_run: bool,
+):
+    """Remove files found in housekeeper bundles"""
 
     housekeeper_api: HousekeeperAPI = context.housekeeper_api
     status_db: Store = context.status_db
 
-    cases: Iterable[models.Family]
-    if case_id:
-        cases: List[models.Family] = [status_db.family(case_id)]
-    else:
-        cases: Iterable[models.Family] = status_db.families()
-    for case in cases:
-        case_id = case.internal_id
-        last_version: Optional[hk_models.Version] = housekeeper_api.last_version(bundle=case_id)
-        if not last_version:
-            continue
-        last_version_file_paths: List[Path] = [
-            Path(hk_file.full_path)
-            for hk_file in housekeeper_api.get_files(
-                bundle=case_id, tags=None, version=last_version.id
+    date_threshold: datetime = datetime.now() - timedelta(days=days_old)
+
+    analyses = status_db.analyses_before_date(
+        case_id=case_id, before=date_threshold, pipeline=pipeline
+    )
+    size_cleaned = 0
+    for analysis in analyses:
+        LOG.info(f"Cleaning analysis {analysis}")
+        bundle_name = analysis.family.internal_id
```

Comment:
Would be great with some more typehints throughout

Reply:
done